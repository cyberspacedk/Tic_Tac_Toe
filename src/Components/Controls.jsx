import React from "react";
import styled from "styled-components";

const ControlsWrapper = styled.div`
  font-size: 1.2rem;
  padding: 30px;
  width: 100vw;
  display: flex;
  justify-content: space-around;
  span {
    font-size: 2rem;
    color: #da00ffcc;
    font-weight: 700;
	}
	button{
		font-size: 1.5rem;
		border-bottom:2px solid #000;
		transition: all .2s linear; 
	}
	button:hover{
		text-shadow: 4px 6px 3px #989898;
		transition: all .2s linear;
	}
`;

const Controls = ({ winner, length, snapshot, switchSnap }) => {
  
const canForward = snapshot !== length;
const canBack = Boolean(snapshot); 

  return (
    <ControlsWrapper>
			<button name="go-back" onClick={()=> canBack && switchSnap(snapshot-1)}>Step Back</button>
      <span>{winner}</span> 
			<button name="go-forward" onClick={()=> canForward && switchSnap(snapshot+1)}>Step Forward</button> 
    </ControlsWrapper>
  );
};

export default Controls;
