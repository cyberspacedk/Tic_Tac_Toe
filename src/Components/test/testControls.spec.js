import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Controls from "../Controls";

describe("Controls component.History actions and show current", () => {
    
// 1- snapshot
// 2- simulate. click back
// 3- simulate. click forward
// 4- simulate. string value

  const mockClick = jest.fn(); 

  const props = {
    winner: "",
    length: 1,
    snapshot: 1,
    switchSnap: mockClick
  };

  const myComponent = shallow(<Controls {...props} />);

  it("Controls snapshot", () => { 
    expect(shallowToJson(myComponent)).toMatchSnapshot();
  });

  it("Simulate. check click back ", () => { 
    myComponent.find('[name="go-back"]').simulate('click');
    expect(mockClick).toHaveBeenCalledTimes(1);
  });

  it("Simulate. check click forward", () => {  
    myComponent.find('[name="go-forward"]').simulate('click');
    expect(mockClick).toHaveBeenCalledTimes(1);
  });
 
  it("Simulate. should show current status (move or winner)", () => {
    const nextProps = {
      ...props,
      winner: "Win Player-1"
    };
    const myComponent = shallow(<Controls {...nextProps} />);
    expect(myComponent.find("span").text()).toEqual(nextProps.winner);
  });

  
});
