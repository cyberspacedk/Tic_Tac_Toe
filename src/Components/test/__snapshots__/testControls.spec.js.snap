// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component.History actions and show current Controls snapshot 1`] = `
<styled.div>
  <button
    name="go-back"
    onClick={[Function]}
  >
    Step Back
  </button>
  <span />
  <button
    name="go-forward"
    onClick={[Function]}
  >
    Step Forward
  </button>
</styled.div>
`;
